### DISCUSSION TITLE
Recommended GUI tool to build a network structure and CPD - to work with PGMPY

### DISCUSSION THREAD

**Discussion by sachinsdesai** [pgmpy/pgmpy#1680] [Q&A]
Hello folks. Has anyone tried building a network with a GUI tool (to make it easier to build and maintain), export it and then have PGMPY read it? I know that pgmpy.readwrite has a lot of readers and writers. I tried building a small; sample network with HUGIN (9.4) and save it to .NET format and read it into PGMPY using pgmpy.readwrite.NETReader but it did not work. I have a simple 4-node network (3 parents, all pointing to a child node) which is saved to a .net file. Here's the error that I get when I try to read using NetReader. Any pointers/suggestions?

```
File ~/environments/py_venvs/pgmpy/lib/python3.8/site-packages/pgmpy/readwrite/NET.py:640, in NETReader.get_values(self)
    638     par_states_prod = prod([len(states[par]) for par in pars])
    639     cpd_flat = np.array(match[0], dtype="float64")
--> 640     cpd_2d = cpd_flat.reshape(par_states_prod, var_state_num).T
    642     variable_cpds[var] = cpd_2d
    644 return variable_cpds

ValueError: cannot reshape array of size 2 into shape (8,2)
```

**Comment by ankurankan** [comment]
@sachinsdesai I am not sure why this is throwing an error. Would it be possible to share your .net file so that I can try to reproduce the error?

**Reply by sachinsdesai** [reply]
Thanks @ankurankan . Will create a trimmed version and send over. In the meantime, any recommendations on what is a good web-based or MAC based tool that I can use to graphically build the Bayesian Network and save it in a format that PGMPY can understand?
